### PR TITLE
ROC-5193 Capture defendant's full name - Defendant Journey

### PR DESCRIPTION
**Pull request by sudheer-ajax**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-5193

### Change description

Capture defendant's full name - Defendant Journey

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [ ] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/test/features/response/routes/your-details.ts [hunk 1]
```diff
@@ -56,6 +76,66 @@ describe('Defendant user details: your name page', () => {
...
+          draftStoreServiceMock.resolveFind('response')
+          claimStoreServiceMock.resolveRetrieveClaimByExternalId()
+
+          await request(app)
+            .get(pagePath)
+            .set('Cookie', `${cookieName}=ABC`)
+            .expect(res => expect(res).to.be.successful.withoutText('Last Name'))
+        })
+
+        it('should render page without firstName when claim doesn\'t have title', async () => {
```
Comment (robertparkinson) [inline src/test/features/response/routes/your-details.ts]: please don't use escape characters. use "does not" instead

File: src/main/public/stylesheets/components/_form.scss [hunk 2]
```diff
@@ -3,6 +3,9 @@
 //noinspection CssUnknownTarget this file is generated by gulp
 @import 'colours';
 
+.title, #title {
```
Comment (robertparkinson) [inline src/main/public/stylesheets/components/_form.scss]: I don't see where title is being used as a class any where?

Comment (sudheer-ajax) [reply src/main/public/stylesheets/components/_form.scss]: we have a input field title in your details and the id of the input is title

Comment (robertparkinson) [reply src/main/public/stylesheets/components/_form.scss]: that will be covered by the #title we don't need the .title that I can see?

File: src/main/features/response/routes/your-details.ts [hunk 3]
```diff
@@ -64,8 +64,12 @@ export default express.Router()
...
-
+      const claim: Claim = res.locals.claim
+      const draft: Draft<ResponseDraft> = res.locals.responseDraft
       if (form.hasErrors()) {
+        if (claim.claimData.defendants[0].firstName) {
+          draft.document.defendantDetails.partyDetails.firstName = claim.claimData.defendants[0].firstName
+        }
         renderView(form, res)
       } else {
         const claim: Claim = res.locals.claim
```
[the remark this answers is not in the document]
Comment (sudheer-ajax) [reply src/main/features/response/routes/your-details.ts]: done

File: src/main/features/response/views/your-details.njk [hunk 4]
```diff
@@ -16,51 +16,57 @@
...
+            ] %}
+          {% endif %}
+
           {% set addressSectionHeading = 'Your address' %}
-        {% elif form.model.type == 'soleTrader' %}
-          {%
-            set fields = [
+          {% elif form.model.type == 'soleTrader' %}
+          {% if claim.claimData.defendants[0].firstName and claim.claimData.defendants[0].firstName !== 'undefined' %}
+            {% set fields = [
```
[the remark this answers is not in the document]
Comment (sudheer-ajax) [reply src/main/features/response/views/your-details.njk]: read only, I have added in missing if condition now

File: src/test/features/response/routes/your-details.ts (hunk 1 above)
Comment (sudheer-ajax) [reply src/test/features/response/routes/your-details.ts]: done

File: src/main/public/stylesheets/components/_form.scss (hunk 2 above)
Comment (sudheer-ajax) [reply src/main/public/stylesheets/components/_form.scss]: agree , just added for future any title / or similar field which doesn't have a id title, I will remove it for now

File: src/main/app/claims/models/details/yours/party.ts [hunk 5]
```diff
@@ -3,18 +3,21 @@ import { PartyType } from 'common/partyType'
 
 export class Party {
   type: string
+  title: string
```
[the remark this answers is not in the document]
Comment (sudheer-ajax) [reply src/main/app/claims/models/details/yours/party.ts]: As discussed, this is a converter from all 4 types to backend data model

File: src/main/app/forms/models/partyDetails.ts [hunk 6]
```diff
@@ -31,6 +31,11 @@ export class PartyDetails {
   @ValidateNested({ groups: ['claimant', 'defendant', 'response'] })
   correspondenceAddress?: CorrespondenceAddress
 
+  firstName?: string
```
[the remark this answers is not in the document]
Comment (sudheer-ajax) [reply src/main/app/forms/models/partyDetails.ts]: Fixed

File: src/main/app/claims/models/details/yours/party.ts [hunk 7]
```diff
@@ -3,18 +3,21 @@ import { PartyType } from 'common/partyType'
...
-              type?: string,
-              name?: string,
-              address?: Address,
-              correspondenceAddress?: Address,
-              mobilePhone?: string, email?: string) {
+    type?: string,
+    name?: string,
+    address?: Address,
+    correspondenceAddress?: Address,
+    mobilePhone?: string, email?: string) {
```
Comment (KiranVarma02) [inline src/main/app/claims/models/details/yours/party.ts]: move email to new line please

Comment (sudheer-ajax) [reply src/main/app/claims/models/details/yours/party.ts]: done

File: src/main/features/response/index.ts [hunk 8]
```diff
@@ -82,13 +64,7 @@ export class Feature {
     app.all(/^\/case\/.+\/response\/(?!confirmation|receipt|summary).*$/,
       DraftMiddleware.requestHandler(new DraftService(), 'response', 100, (value: any): ResponseDraft => {
         return new ResponseDraft().deserialize(value)
-      }),
-      (req: express.Request, res: express.Response, next: express.NextFunction) => {
-        res.locals.draft = res.locals.responseDraft
```
Comment (robertparkinson) [inline src/main/features/response/index.ts]: I think we still need this bit. other wise a lot of the pages won't have the correct draft I believe?

File: src/main/features/response/index.ts [hunk 9]
```diff
@@ -82,13 +64,7 @@ export class Feature {
...
-        next()
-      },
-      initiatePartyFromClaimHandler
```
Comment (robertparkinson) [inline src/main/features/response/index.ts]: it's just this bit that needs removing

**Pull request merged**